ProcessMaker/processmaker
# FOUR-13028 | The Editing of Any Process Can Be Accessed With Any User Who Has Projects Permissions Activated

# Issue
Ticket: [FOUR-13028](https://processmaker.atlassian.net/browse/FOUR-13028)

This pull request addresses an issue with Project Permissions that allowed unauthorized access and modification of processes, scripts, and screens.

# Solution
Updated AuthServiceProvider's boot function. Queried the database for projects the user is a member or owner of and retrieved associated project assets. Adjusted `$allowedEndpoints` for Processes, Scripts, and Screens to limit user access to endpoints for assets associated with their projects.

# How to Test
1. Go to branch `observation/FOUR-13028` in `processmaker`.
2. Have Processes, Screens, and Scripts created in your environment.
3. Create a User with only Projects permissions.
4. Login with that User. Have a Project created with a Process, Screen, and Script.
4. Change the URL to access process editing: `/modeler/{processId}`.
5. Edit and Publish the process.
	- Users should only be able to access and edit processes that are associated with their projects.
	- If you visit the Modeler for a process associated with the user's project, you should be able to see and edit the process. If you visit the Modeler with a process that is not part of one of the user's projects, you should see a 'Not Authorized' page.
6. Repeat these steps with the Screen and Script editors.

ci:next

# Code Review Checklist


[FOUR-13028]: https://processmaker.atlassian.net/browse/FOUR-13028?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
